### PR TITLE
Added support for array metric datatypes

### PR DESCRIPTION
**Pull request by wes-johnson**
Signed-off-by: wes-johnson <wes0johnson@gmail.com>

### CODE REVIEW

**Review by nathandavenport — APPROVED** [review]

**Review by ckienle — APPROVED** [review]

File: client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadDecoder.java [hunk 1]
```diff
@@ -369,7 +448,7 @@ private Object getParameterValue(SparkplugBProto.Payload.Template.Parameter prot
 				}
 			case Unknown:
 			default:
-				logger.error("Unknown DataType: " + protoType);
+				logger.error("Unknown DataSetDataType: " + protoType);
```
Comment (ibinshtok) [inline client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadDecoder.java:451]: Log formatting

File: client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java [hunk 2]
```diff
@@ -221,7 +223,7 @@ public SparkplugBPayloadEncoder() {
 						break;
 					case Unknown:
 					default:
-						logger.error("Unknown DataType: " + value.getType());
+						logger.error("Unknown PropertyDataType: " + value.getType());
```
Comment (ibinshtok) [inline client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java:226]: Log formatting

File: client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java [hunk 3]
```diff
@@ -235,51 +237,51 @@ public SparkplugBPayloadEncoder() {
...
-				}
-				break;
-			case Unknown:
-			default:
-				logger.error("Unknown Type: " + type);
-				throw new Exception("Failed to encode");
+					break;
+				case Unknown:
+				default:
+					logger.error("Unknown Type: " + type);
```
Comment (ibinshtok) [inline client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java:281]: Log formatting

File: client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java [hunk 4]
```diff
@@ -420,9 +422,97 @@ public SparkplugBPayloadEncoder() {
...
+							}
+						}
+						booleanByteBuffer.put(nextByte);
+					}
+					metricBuilder.setBytesValue(ByteString.copyFrom(booleanByteBuffer.array()));
+					break;
 				case Unknown:
 				default:
-					logger.error("Unknown DataType: " + metric.getDataType());
+					logger.error("Unsupported MetricDataType: " + metric.getDataType());
```
Comment (ibinshtok) [inline client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java:515]: Log formatting

File: client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java [hunk 5]
```diff
@@ -543,7 +633,7 @@ public SparkplugBPayloadEncoder() {
 				}
 				break;
 			default:
-				logger.error("Unknown DataType: " + value.getType());
+				logger.error("Unknown DataSetDataType DataType: " + value.getType());
```
Comment (ibinshtok) [inline client_libraries/java/src/main/java/org/eclipse/tahu/message/SparkplugBPayloadEncoder.java:636]: Log Formatting

**Review by ibinshtok — APPROVED** [review]

**Pull request merged**